nntrainer/layers/gru.h

```diff
@@ -106,54 +93,34 @@ class GRULayer : public LayerImpl {
   inline static const std::string type = "gru";
 
 private:
-  std::tuple<props::Unit>
-    props; /**< lstm layer properties : unit - number of output neurons */
-  std::array<unsigned int, 7> wt_idx; /**< indices of the weights */
-
   /**
-   * @brief     activation type for hidden state : default is sigmoid
-   */
-  ActivationType hidden_state_activation_type;
+   * Unit: number of output neurons
+   * HiddenStateActivation: activation type for hidden state. default is tanh
+   * RecurrentActivation: activation type for recurrent. default is sigmoid
+   * ReturnSequence: option for return sequence
+   * DropOutRate: dropout rate
+   *
+   * */
+  std::tuple<props::Unit, props::HiddenStateActivation,
+             props::RecurrentActivation, props::ReturnSequences,
+             props::DropOutRate>
+    gru_props;
+  std::array<unsigned int, 7> wt_idx; /**< indices of the weights */
 
   /**
    * @brief     activation function for h_t : default is sigmoid
    */
   ActiFunc acti_func;
 
-  /**
-   * @brief     activation type for recurrent : default is tanh
-   */
-  ActivationType recurrent_activation_type;
-
   /**
    * @brief     activation function for recurrent : default is tanh
    */
   ActiFunc recurrent_acti_func;
 
-  /**
-   * @brief     variable to set return sequences
-   */
-  bool return_sequences;
-
-  /**
-   * @brief     drop out rate
-   */
-  float dropout_rate;
-
   /**
    * @brief     to pretect overflow
    */
   float epsilon;
```

Comment:
using epsilon property like in batch norm layer?

Reply:
Does the epsilon is property of gru layer? I think epsilon is just a small constant value to check activate dropout in the rnn, lstm, gru layer.